### PR TITLE
DEVOPS-28 Add OAuth-proxy

### PR DESCRIPTION
**Pull request by anton-delphai**
Run OAuthProxy sidecar when requested

### CODE REVIEW

File: charts/delphai-keda/templates/deployment.yml [hunk 1]
```diff
@@ -71,3 +72,68 @@ spec:
               containerPort: {{ .Values.http.port }}
             - name: h2c
               containerPort: {{ .Values.grpc.port }}
+
+        {{ if and .Values.http.enabled .Values.http.public .Values.http.publicWithOAuth }}
+        - name: oauth2-proxy
+          image: "quay.io/oauth2-proxy/oauth2-proxy:v7.2.0"
+          args:
+            - '--email-domain=*'
```
Comment (anton-delphai) [inline charts/delphai-keda/templates/deployment.yml:80]: This should work being passed as ENV, but it doesn't

File: charts/delphai-keda/templates/deployment.yml [hunk 2]
```diff
@@ -71,3 +72,68 @@ spec:
...
+            valueFrom:
+              secretKeyRef:
+                name: oauth2-proxy
+                key: cookie-secret
+
+          - name: OAUTH2_PROXY_SKIP_AUTH_ROUTES
+            value: "^/favicon\\.ico$"
+
+          - name: OAUTH2_PROXY_UPSTREAMS
+            value: "http://127.0.0.1:{{ .Values.http.port }}"
```
Comment (anton-delphai) [inline charts/delphai-keda/templates/deployment.yml:122]: This should work being passed as command argument, but it doesn't — it silently ignores. For some weird reason it's not even a required parameter

File: charts/delphai-keda/templates/deployment.yml [hunk 3]
```diff
@@ -71,3 +72,68 @@ spec:
...
+          - name: OAUTH2_PROXY_HTTP_ADDRESS
+            value: 0.0.0.0:{{ .Values.http.oAuthProxyPort }}
+
+          - name: OAUTH2_PROXY_PROVIDER
+            value: "oidc"
+
+          - name: OAUTH2_PROXY_OIDC_ISSUER_URL
+            valueFrom:
+              secretKeyRef:
+                name: oauth2-proxy
```
Comment (anton-delphai) [inline charts/delphai-keda/templates/deployment.yml:91]: I've created this secret manually in all clusters, I'll create a ticket to generate it with TF

**Comment by anton-delphai** [comment]
Now it's working here: https://ml-tools.delphai.pink

File: charts/delphai-keda/templates/deployment.yml (hunk 1 above)
Comment (eugene-delphai) [reply charts/delphai-keda/templates/deployment.yml:80]: :(

**Review by eugene-delphai — APPROVED** [review]

File: charts/delphai-keda/templates/deployment.yml (hunk 1 above)
Comment (anton-delphai) [reply charts/delphai-keda/templates/deployment.yml:80]: Yeah, too a lot of time to debug that — all other params were working fine

**Pull request merged**